### DISCUSSION TITLE
Multi-viewport editing

### DISCUSSION THREAD

**Discussion by AMZN-Liv** [o3de/rfcs#26] [Ideas]
**Is your feature request related to a problem? Please describe.**
 Today, users can only work on one level or asset at a time, which limits workflows that rely on asset dependencies and making changes to components that may be reflected in other levels. 

**Describe the solution you'd like**
Multi-viewport editing would make it possible to open multiple levels and prefabs at the same time with different viewports so that changes could be made and propagate between the open views.

**Comment by arvinmoses** [comment]
2nding this.  On a few different engines I've used, they ran it with this feature.  Helps with copying and pasting between scenes, but more importantly for having zoos and creating approved sets of assets for a particular environment

**Reply by arvinmoses** [reply]
Replying to this in response to Jonny Galloway's suggestion to include user needs from discord:

I'd specifically like the ability to do multiple scenes open.  This allows a few different things that I've used in a game pipe and I feel is really useful:

1. Ability to make an asset zoo.  This allows leads and ad to construct a scene with approved assets.  Users can have both the zoo and work scene open simultaneously.  User would select an asset in the zoo, hit ctrl + c, then paste in the scene with ctrl + v.
2. Ability to have two similar scenes open at once.  Users can then take a group of assets that work for solving a visual problem in one scene and very easily copy and past in the other.

**Comment by WashedUpStudio** [comment]
This would be a great addition, I hope to see this past the ideas phase.
edit; I just got word this is already possible since atom supports multiple views.

**Comment by Infurael** [comment]
The feature to have multiple viewports for editing has been discussed in the team that's working on the new Prefab system. Since the original editor is fully centered around a single viewport there's a lot of work that still needs to happen. A big one has been multiple viewport support in Atom, which we have. The next technical hurdle is removing the Entity Context EBuses as they limit flexibility and replace them with an Entity Context in AzFramework::Scene. @sconel or @monroegm can maybe talk to the status of that. Beyond that there's of course more work to be done on adding support to the editor itself and addressing issues with singleton components.

The main driver for this feature has been the ability to do side-by-side comparison. For instance to have one prefab build with network support in one view and another view where the same prefab is build without network support so tweaks for various network latency scenarios can be made. The suggestion @arvinmoses posted is another great example of why this feature is needed. Would @Amzn-Roby and I would love to see other scenarios as well, so post them here if you can think of any more.
